SPID/include/LinkedDistribution.h

```diff
@@ -0,0 +1,256 @@
+		void ForEachLinkedDeathDistributionSet(const DistributedForms& linkedForms, Scope, std::function<void(DistributionSet&)> distribute);
+
+		LinkedForms<RE::SpellItem>      spells{ RECORD::kSpell };
+		LinkedForms<RE::BGSPerk>        perks{ RECORD::kPerk };
+		LinkedForms<RE::TESBoundObject> items{ RECORD::kItem };
+		LinkedForms<RE::TESBoundObject> deathItems{ RECORD::kDeathItem };
+		LinkedForms<RE::TESShout>       shouts{ RECORD::kShout };
+		LinkedForms<RE::TESLevSpell>    levSpells{ RECORD::kLevSpell };
+		LinkedForms<RE::TESForm>        packages{ RECORD::kPackage };
+		LinkedForms<RE::BGSOutfit>      outfits{ RECORD::kOutfit };
+		LinkedForms<RE::BGSOutfit>      sleepOutfits{ RECORD::kSleepOutfit };
+		LinkedForms<RE::BGSKeyword>     keywords{ RECORD::kKeyword };
+		LinkedForms<RE::TESFaction>     factions{ RECORD::kFaction };
+		LinkedForms<RE::TESObjectARMO>  skins{ RECORD::kSkin };
+
+		/// <summary>
+		/// Iterates over each type of LinkedForms and calls a callback with each of them.
+		/// </summary>
+		template <typename Func, typename... Args>
+		void ForEachLinkedForms(Func&& func, Args&&... args);
+	};
+
+#pragma region Implementation
+
+	template <class Form>
+	Form* detail::LookupLinkedForm(RE::TESDataHandler* const dataHandler, INI::RawLinkedForm& rawForm)
+	{
+		using namespace Forms::Lookup;
+
+		try {
+			return Forms::detail::get_form<Form>(dataHandler, rawForm.formOrEditorID, rawForm.path);
+		} catch (const UnknownFormIDException& e) {
+			buffered_logger::error("\t\t[{}] LinkedForm [0x{:X}] ({}) SKIP - formID doesn't exist", e.path, e.formID, e.modName.value_or(""));
+		} catch (const UnknownPluginException& e) {
+			buffered_logger::error("\t\t[{}] LinkedForm ({}) SKIP - mod cannot be found", e.path, e.modName);
+		} catch (const InvalidKeywordException& e) {
+			buffered_logger::error("\t\t[{}] LinkedForm [0x{:X}] ({}) SKIP - keyword does not have a valid editorID", e.path, e.formID, e.modName.value_or(""));
+		} catch (const KeywordNotFoundException& e) {
+			if (e.isDynamic) {
+				buffered_logger::critical("\t\t[{}] LinkedForm {} FAIL - couldn't create keyword", e.path, e.editorID);
+			} else {
+				buffered_logger::critical("\t\t[{}] LinkedForm {} FAIL - couldn't get existing keyword", e.path, e.editorID);
+			}
+		} catch (const UnknownEditorIDException& e) {
+			buffered_logger::error("\t\t[{}] LinkedForm ({}) SKIP - editorID doesn't exist", e.path, e.editorID);
+		} catch (const MalformedEditorIDException& e) {
+			buffered_logger::error("\t\t[{}] LinkedForm (\"\") SKIP - malformed editorID", e.path);
+		} catch (const MismatchingFormTypeException& e) {
+			std::visit(overload{
+						   [&](const FormModPair& formMod) {
+							   auto& [formID, modName] = formMod;
+							   buffered_logger::error("\t\t[{}] LinkedForm [0x{:X}] ({}) SKIP - mismatching form type (expected: {}, actual: {})", e.path, *formID, modName.value_or(""), RE::FormTypeToString(e.expectedFormType), RE::FormTypeToString(e.actualFormType));
```

Comment:
RE::FormTypeToString is not needed in logging, since FormType is automatically overloaded for std::format